cpp/ql/src/semmle/code/cpp/File.qll

```diff
@@ -358,6 +358,11 @@ class File extends Container, @file {
   string getShortName() { files(underlyingElement(this), _, result, _, _) }
 }
 
+/**
+ * Holds if any file was compiled by a Microsoft compiler.
+ */
+cached predicate anyFileCompiledAsMicrosoft() { any(File f).compiledAsMicrosoft() }
```

Comment:
Adding a cached stage just for this seems heavy-handed. Isn't it enough to write `pragma[noinline]`?

Reply:
That doesn't seem to be enough, but I've found a solution improving `compiledAsMicrosoft()` and removing `cached`.